libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java

```diff
@@ -446,18 +446,64 @@ public void onClick(DialogInterface dialog,
      * a custom fingerprint auth layout if the app chooses to do so.
      */
     protected void showFingerprintDialog() {
-        if (passcodeManager != null && isFingerprintEnabled()) {
-            final FingerprintAuthDialogFragment fingerprintAuthDialog = new FingerprintAuthDialogFragment();
-            fingerprintAuthDialog.setContext(this);
-            fingerprintAuthDialog.show(getFragmentManager(), "fingerprintDialog");
+        final FingerprintAuthDialogFragment fingerprintAuthDialog = new FingerprintAuthDialogFragment();
+        fingerprintAuthDialog.setContext(this);
+        fingerprintAuthDialog.show(getFragmentManager(), "fingerprintDialog");
+    }
+
+    /**
+     * Displays the dialog provided by the OS for biometric authentication
+     * using {@link android.hardware.biometrics.BiometricPrompt}.
+     */
+    @TargetApi(VERSION_CODES.P)
+    protected void showBiometricDialog() {
+
+        /*
+         * TODO: Remove this check once minAPI >= 28.
+         */
+        if (VERSION.SDK_INT >= VERSION_CODES.P) {
```

Comment:
Here's the new `Android P` stuff. The actions are the same as the old fingerprint code.